File: 2025/day11.py
from collections import deque


def count_paths(n, edges, source, destination):

    # Create adjacency list (1-based indexing)
    graph = [[] for _ in range(n + 1)]
    in_degree = [0] * (n + 1)

    for u, v in edges:
        graph[u].append(v)
        in_degree[v] += 1

    q = deque()
    for i in range(1, n + 1):
        if in_degree[i] == 0:
            q.append(i)

    topological_order = []
    while q:
        node = q.popleft()
        topological_order.append(node)

        for neighbor in graph[node]:
            in_degree[neighbor] -= 1
            if in_degree[neighbor] == 0:
                q.append(neighbor)

    ways = [0] * (n + 1)
    ways[source] = 1

    for node in topological_order:
        for neighbor in graph[node]:
            ways[neighbor] += ways[node]

    return ways[destination]


with open("2025/input/day11") as f:
    data = f.read().splitlines()

    nodes = {}

    for row in data:
        start, end = row.split(": ")
        successors = end.split(" ")
        nodes[start] = successors

    id_nodes = {}
    i = 1
    for k in nodes.keys():
        id_nodes[k] = i
        i += 1

    id_nodes["out"] = i

    graph = []
    for k, v in nodes.items():
        for e in v:
            graph.append([id_nodes[k], id_nodes[e]])

    count = count_paths(len(id_nodes), graph, id_nodes["you"], id_nodes["out"])
    print("Partie 1:", count)

    count1 = count_paths(len(id_nodes), graph, id_nodes["svr"], id_nodes["fft"])
    count2 = count_paths(len(id_nodes), graph, id_nodes["fft"], id_nodes["dac"])
    count3 = count_paths(len(id_nodes), graph, id_nodes["dac"], id_nodes["out"])
    count = count1 * count2 * count3
    print("Partie 2:", count)

File: 2025/rust/src/aoc2025/day11.rs
use rdcl_aoc_helpers::input::WithReadLines;
use std::fs::File;
use std::collections::HashMap;


fn count_paths(n: u32, edges: &Vec<Vec<u32>>, source: u32, destination: u32) -> u64 {
    
}

pub fn day11(path: &String) {
    let input: Vec<String> = File::open(path).read_lines::<String>(1).collect();

    let mut nodes:HashMap<String, Vec<String>> = HashMap::new();
    let mut id_nodes:HashMap<String, u32> = HashMap::new();
    let mut id = 1;

    for row in &input {
        let s1 = row.split_once(": ").unwrap();
        let start = s1.0;
        let s2 = s1.1.split(" ").map(|x| x.to_string()).collect::<Vec<String>>();
        nodes.insert(start.to_string(), s2);
    }
    for key in nodes.keys() {
        id_nodes.insert(key.to_string(), id);
        id += 1;
    }
    id_nodes.insert( "out".to_string(), id);

    let mut graph:Vec<Vec<u32>> = Vec::new();

    for item in &nodes {
        for node in item.1 {
            graph.push(vec![id_nodes.get(item.0).unwrap().clone(), id_nodes.get(node).unwrap().clone()]);
        }
    }

    println!("{:?}", graph);
    println!("Partie 1: {}", 0);
    println!("Partie 2: {}", 0);
}